x-pack/plugins/advanced_ui_actions/public/components/action_wizard/action_wizard.tsx

```diff
@@ -16,40 +16,23 @@ import {
 } from '@elastic/eui';
 import { txtChangeButton } from './i18n';
 import './action_wizard.scss';
+import { ActionFactory } from '../../dynamic_actions';
 
-// TODO: this interface is temporary for just moving forward with the component
-// and it will be imported from the ../ui_actions when implemented properly
-// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
-export type ActionBaseConfig = {};
-export interface ActionFactory<Config extends ActionBaseConfig = ActionBaseConfig> {
-  type: string; // TODO: type should be tied to Action and ActionByType
-  displayName: string;
-  iconType?: string;
-  wizard: React.FC<ActionFactoryWizardProps<Config>>;
-  createConfig: () => Config;
-  isValid: (config: Config) => boolean;
-}
-
-export interface ActionFactoryWizardProps<Config extends ActionBaseConfig> {
-  config?: Config;
-
-  /**
-   * Callback called when user updates the config in UI.
-   */
-  onConfig: (config: Config) => void;
-}
+type ActionBaseConfig = object;
+type ActionFactoryBaseContext = object;
```

Comment:
I have removed those two types, they were internal helper types of that `<ActionWizard>` React component. But that react component knows nothing about those variables&mdash;and that is by design&mdash;it simply passes those variables to drilldowns. From the perspective of `<ActionWizard>` component those are simply *opaque* objects that it needs to pass down to drilldowns.

> can we merge that into advanced_ui_actions to avoid the introduction of the two apis, `registerDrilldown` and `registerActionFactory`?

Yes, I've moved `registerDrilldown` to actions plugin and made `registerActionFactory` private.

> This is a lot of different context types to follow so some simple examples of which is what would be helpful.

There are two context types: (1) factory context; (2) execution context. We now have 4 examples where that is illustrated:

- Dashboard drilldown in `x-pack/dashboard_enhanced`
- 3 sample drilldowns in `x-pack/examples/ui_actions_enhanced`

Reply:
`PlaceContext` and `CreationContext` are just aliases for the same thing, I've renamed `CreationContext` to place `PlaceContext`.